internal/datacoord/compaction_task_clustering.go

```diff
@@ -700,12 +700,17 @@ func (t *clusteringCompactionTask) doCompact() error {
 	}
 	err = t.sessions.Compaction(context.Background(), t.GetTaskProto().GetNodeID(), t.GetPlan())
 	if err != nil {
-		if errors.Is(err, merr.ErrDataNodeSlotExhausted) {
```

Comment:
Why is the logic for checking ErrDataNodeSlotExhausted no longer needed?

Reply:
Because only pipelining && NullNodeID can be reassigned, just set NullNodeID makes this task un-scheduleable forever.